templates/terraform/custom_flatten/repository_short_name_from_name.go.erb

```diff
@@ -0,0 +1,23 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2017 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData) interface{} {
+    if v == nil {
+        return v
+    }
+
+  // We can't use a standard name_from_self_link because the name can include /'s
+  parts := strings.SplitAfterN(v.(string), "/", 4)
+	return parts[3]
```

Comment:
nit: gofmt fixes it so up to you if you feel like changing it, but there's some tabs/spaces inconsistency going on